core/src/main/scala/reftree/diagram/Animation.scala

```diff
@@ -83,6 +83,32 @@ object Animation {
   /** Create an animation builder from a sequence of starting values */
   def startWithSequence[A: ToRefTree](start: Seq[A]) = Builder(start.toVector)
 
+  private implicit class TakeUntilStream[T](s: Stream[T]) {
+    def takeWhileInclusive(p: T => Boolean): Stream[T] =
+      if (!s.isEmpty && p(s.head)) Stream.cons(s.head, s.tail takeWhileInclusive p)
+      else Stream.cons(s.tail.head, Stream.empty)
+  }
+
+  /** Create an animation builder from a seed until a fixpoint (f(x) == x) is reached */
+  def iterateUntilFixPoint[A: ToRefTree](seed: A)(iteration: (A => A))(
```

Comment:
You mean a method to add a single frame to a builder? Like `def add(frame: A)`? Sure, sounds good (although if you want to start a new animation, you can also create two animations and chain them, but that would mean more work if you want to share the `build` arguments).